extensions/wikia/ImageServing/imageServing.class.php

```diff
@@ -125,11 +125,15 @@ public function getImages( $limit = 5, $driverName = null ) {
 
 			// fetch articles metadata (try from cache)
 			foreach ( $articles as $key => $value ) {
-				$mcValue = $this->memc->get( $this->makeKey( $key, Article::newFromID( $key )->getRevIdFetched() ) );
+				$article = Article::newFromID( $key );
 
-				if ( !empty( $mcValue ) ) {
-					unset( $articles[$key] );
-					$this->addArticleToList( $mcValue );
+				if ( $article instanceof  Article ) {
+					$mcValue = $this->memc->get( $this->makeKey( $key, $article->getRevIdFetched() ) );
```

Comment:
It seems to me we can bypass instantiating `Article` here and just use `Title::newFromID` + `Title::getLatest`